lib/mixlib/shellout.rb

```diff
@@ -322,7 +363,8 @@ def parse_options(opts)
           self.log_tag = setting
         when 'environment', 'env'
           self.environment = setting || {}
-
+        when 'with_logon'
```

Comment:
Since we're actually creating a new public argument here could we rename this to `login_shell: true`?  or just `login: true`?  We don't need to maintain the clunky name we picked for the internal variable.